typescript/ws-protocol/src/types.ts

```diff
@@ -40,12 +40,22 @@ export type Channel = {
   schema: string;
   schemaEncoding?: string;
 };
+
+type ServiceRequestDefinition = {
+  encoding: string;
+  schemaName: string;
+  schemaEncoding: string;
+  schema: string;
+};
+type ServiceResponseDefinition = ServiceRequestDefinition;
 export type Service = {
   id: number;
   name: string;
   type: string;
-  requestSchema: string;
-  responseSchema: string;
+  request?: ServiceRequestDefinition; // Must be given if requestSchema is not given.
+  response?: ServiceResponseDefinition; // Must be given if responseSchema is not given.
+  requestSchema?: string; // Must be given if request is not given.
+  responseSchema?: string; // Must be given if response is not given.
```

Comment:
Should we also mark these as deprecated, or maybe not since we will probably keep supporting it for a long time?